src/ansys_sphinx_theme/extension/linkcode.py

```diff
@@ -0,0 +1,269 @@
+    app : sphinx.application.Sphinx
+        The Sphinx application instance.
+    doctree : docutils.nodes.Node
+        The document tree for the current page.
+
+    Notes
+    -----
+    The function uses the `sphinx_linkcode_resolve` function to resolve the link for each
+    documented object. The link is added as a "View Source" link to the documentation
+    when rendered.
+
+    For Python, the function extracts information such as module name and fullname from
+    the signature node to create the link.
+
+    The `link_code_library` configuration option can be used to specify a custom library
+    for link resolution.
+
+    Notes
+    -----
+    Adapted from the original `linkcode` extension in Sphinx, see
+    https://github.com/sphinx-doc/sphinx/blob/main/sphinx/ext/linkcode.py
+    """
+    env = app.builder.env
+    html_context = getattr(env.config, "html_context")
+    github_user = html_context.get("github_user", "")
+    github_repo = html_context.get("github_repo", "")
+    github_source = html_context.get("source_path", "")
+    github_version = html_context.get("github_version", "main")
+    if github_user and github_repo:
+        library = f"{github_user}/{github_repo}"
+    elif hasattr(env.config, "link_code_library") and hasattr(env.config, "link_code_source"):
+        library = getattr(env.config, "link_code_library")
+        github_source = getattr(env.config, "link_code_source")
+        github_version = getattr(env.config, "link_code_branch")
+
+    else:
+        raise AttributeError("The library should have either html_context or link_code_library.")
+    if not library:
+        raise AttributeError(
+            "The conf.py file either should have html_context or link_code_library with github_repo and github_user."  # noqa: E501
+        )
+
+    for objnode in list(doctree.findall(addnodes.desc)):
+        domain = objnode.get("domain")
+        uris: set[str] = set()
+        for signode in objnode:
+            if not isinstance(signode, addnodes.desc_signature):
+                continue
+
+            # Convert signode to a specified format
+            info = {}
+            for key in DOMAIN_KEYS.get(domain, []):
+                value = signode.get(key)
+                if not value:
+                    value = ""
+                info[key] = value
+            if not info:
+                continue
+
+            # Call user code to resolve the link
+            uri = sphinx_linkcode_resolve(
+                domain=domain,
+                info=info,
+                library=library,
+                source_path=github_source,
+                github_version=github_version,
+            )
+            if not uri:
+                # no source
+                continue
+
+            if uri in uris or not uri:
+                # only one link per name, please
+                continue
+            uris.add(uri)
+
+            inline = nodes.inline("", _("[source]"), classes=["viewcode-link"])
```

Comment:
This is where we used the '_' from sphinx.locale, in which we creates an inline node, which is used to represent inline content in the document. The content of this node is "[source]", and it's wrapped in an underscore _ function, which is typically used for translation (localization) of strings. i tried with get_translation method, but i am not able to get the exact source name. https://www.sphinx-doc.org/en/master/extdev/i18n.html#sphinx.locale.get_translation this documentation might be helpful @jorgepiloto and @RobPasMue  continuation of https://github.com/ansys/ansys-sphinx-theme/pull/269/files/deeaa53ab257f3541c250a3520b1fa61f9d15047#r1286783080
![image](https://github.com/ansys/ansys-sphinx-theme/assets/104772255/577f26e2-b32c-4113-a285-9e711d0f8262)

Reply:
Thanks for the detailed explanation. It is still a bit shocking to me that this is named like `_`.